src/hmacKey.ts

```diff
@@ -0,0 +1,330 @@
+       *       return;
+       *     }
+       *     console.log(hmacKeyMetadata);
+       *   });
+       *
+       * //-
+       * // If the callback is omitted, a promise is returned.
+       * //-
+       * storage.hmacKey('ACCESS_ID')
+       *   .getMetadata()
+       *   .then((data) => {
+       *     const hmacKeyMetadata = data[0];
+       *     console.log(hmacKeyMetadata);
+       *   });
+       */
+      getMetadata: true,
+      /**
+       * @typedef {object} SetHmacKeyMetadata Subset of {@link HmacKeyMetadata} to update.
+       * @property {string} state New state of the HmacKey. Either 'ACTIVE' or 'INACTIVE'.
+       * @property {string} [etag] Include an etag from a previous get HMAC key request
+       *    to perform safe read-modify-write.
+       */
+      /**
+       * @typedef {object} SetHmacKeyMetadataOptions
+       * @property {string} [userProject] This parameter is currently ignored.
+       */
+      /**
+       * @callback HmacKeyMetadataCallback
+       * @param {?Error} err Request error, if any.
+       * @param {HmacKeyMetadata} metadata The updated {@link HmacKeyMetadata} object.
+       * @param {object} apiResponse The full API response.
+       */
+      /**
+       * @typedef {array} HmacKeyMetadataResponse
+       * @property {HmacKeyMetadata} 0 The updated {@link HmacKeyMetadata} object.
+       * @property {object} 1 The full API response.
+       */
+      /**
+       * Updates the state of an HMAC key. See {@link SetHmacKeyMetadata} for
+       * valid states.
+       *
+       * @method HmacKey#setMetadata
+       * @param {SetHmacKeyMetadata} metadata The new metadata.
+       * @param {SetHmacKeyMetadataOptions} [options] Configuration options.
+       * @param {HmacKeyMetadataCallback} [callback] Callback function.
+       * @returns {Promise<HmacKeyMetadataResponse>}
+       *
+       * @example
+       * const {Storage} = require('@google-cloud/storage');
+       * const storage = new Storage();
+       *
+       * const metadata = {
+       *   state: 'INACTIVE',
+       * };
+       *
+       * storage.hmacKey('ACCESS_ID')
+       *   .setMetadata(metadata, (err, hmacKeyMetadata) => {
+       *     if (err) {
+       *       // The request was an error.
+       *       console.error(err);
+       *       return;
+       *     }
+       *     console.log(hmacKeyMetadata);
+       *   });
+       *
+       * //-
+       * // If the callback is omitted, a promise is returned.
+       * //-
+       * storage.hmacKey('ACCESS_ID')
+       *   .setMetadata(metadata)
+       *   .then((data) => {
+       *     const hmacKeyMetadata = data[0];
+       *     console.log(hmacKeyMetadata);
+       *   });
+       */
+      setMetadata: {
+        reqOpts: {
+          method: 'PUT',
+        },
+      },
+    };
+
+    super({
+      parent: storage,
+      baseUrl: `/projects/${storage.projectId}/hmacKeys`,
+      id: accessId,
+      methods,
+    });
+
+    this.accessId = accessId;
```

Comment:
Yes. I'm mirroring what we do with Bucket id (name):

`this.name = name;`

Reply:
Ah, ok. You can skip it here. We do that because we need that name at different places in the code, if I remember correctly. But we don't advertise it as being publicly accessible.

Reply:
Ah gotcha!